pkg/mover/chart.go

```diff
@@ -230,15 +224,65 @@ func (cm *ChartMover) Print() {
 	cm.printMove()
 }
 
-func (cm *ChartMover) loadChart(chartPath string) error {
+func (cm *ChartMover) loadChart(src *Source) error {
 	var err error
+	if src.Chart.Local != nil {
+		cm.chart, err = loader.Load(src.Chart.Local.Path)
+		if err != nil {
+			return &ChartLoadingError{Path: src.Chart.Local.Path, Inner: err}
+		}
+		return nil
+	} else if src.Chart.IntermediateBundle != nil {
+		return cm.loadChartFromIntermediateBundle(src.Chart.IntermediateBundle.Path)
+	}
+	return fmt.Errorf("must provide either a local chart or an intermediate bundle as input")
+}
+
+func (cm *ChartMover) loadChartFromIntermediateBundle(bundlePath string) error {
+	var err error
+	cm.bundle, err = openBundle(bundlePath)
+	if err != nil {
+		return err
+	}
+
+	chartPath, err := os.MkdirTemp("", "bundle-chart-*")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary directory to extract bundled chart %s: %w",
+			cm.bundle.bundlePath, err)
+	}
+	defer os.RemoveAll(chartPath)
+
+	if err := cm.bundle.ExtractChartTo(chartPath); err != nil {
+		return err
+	}
+
 	cm.chart, err = loader.Load(chartPath)
 	if err != nil {
 		return &ChartLoadingError{Path: chartPath, Inner: err}
 	}
 	return nil
 }
 
+func (cm *ChartMover) loadHints(src *Source) error {
+	var err error
+	if src.Chart.IntermediateBundle != nil {
+		if src.ImageHintsFile != "" {
+			return fmt.Errorf("intermediate bundles already embed the hints file" +
+				", skip explicit hints, they will be ignored")
+		}
+		cm.rawHints, err = cm.bundle.LoadHints(cm.logger)
```

Comment:
what about returning the error here? I know that it will be more code but I think it's more idiomatic and will not require the error variable declaration above.

Reply:
checking...

Reply:
`		cm.rawHints, err := cm.intermediateBundle.loadImageHints(cm.logger)`
This does not work:
~~~go
pkg/mover/chart.go:280:3: expected identifier on left side of := (typecheck)
                cm.rawHints, err := cm.intermediateBundle.loadImageHints(cm.logger)
~~~
The error message is not very good, but the problem is simple, the err is declared, but `cm.rawHints` is only assigned, and Go does not seem to allow mixing.

Maybe I can fix some other way.

Reply:
Not super nice but maybe a bit better:
~~~go
		if rawHints, err := cm....; err != nil {
			return err
		} else {
			cm.rawHints = rawHints
		}
~~~

Reply:
Maybe we can do better using load methods instead, like with the chart loading.

Reply:
I think it is better now